internal/session/service_authorize_connection.go

```diff
@@ -0,0 +1,27 @@
+package session
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+// AuthorizeConnection will check to see if a connection is allowed.
```

Comment:
I would start this comment off with:

```
// AuthorizeConnection is a domain service function that ....
```

I think this is a good place to describe the checks that are performed in determining if a Connection is authorized.